packages/ethereum/contracts/src/node-stake/permissioned-module/CapabilityPermissions.sol

```diff
@@ -0,0 +1,760 @@
+ *
+ * It is specifically tailored for interaction with HoprChannels and HoprToken contracts. 
+ * Additionally, it enables the transfer of native tokens to designated addresses, 
+ * while restricting the invocation of payable functions.
+ * 
+ * Some difference between this library and the original `Permissions.sol` contract are:
+ * - This library is designed to support a single role
+ * - No `DelegateCall` is allowed
+ * - Target must be one of the three types: Token, Channels, SEND
+ * - Only scoped functions are allowed. No more wildcard
+ * - Calling payable function is not allowed.
+ * - When calling HoprChannels contracts, permission is check with multiple parameters together
+ * - For Channels targets, the default permission is ALLOWED. However, the default value for other targets is BLOCKED.
+ * - Permissions are not stored bitwise in `scopeConig` (uint256) due to lack of customization
+ * - Utility functions, such as `packLeft`, `packRight`, `unpackFunction`, `unpackParameter`, `checkExecutionOptions` are removed
+ * - Specific helper functions, such as `pluckOneStaticAddress`, `pluckTwoStaticAddresses`, `pluckDynamicAddresses`,  `pluckSendPayload` are derived from `pluckStaticValue` and `pluckDynamicValue`
+ * - helper functions to encode array of function signatures and their respective permissions are added.
+ *
+ * @notice Due to the deployed HoprToken.sol imports OpenZeppelin contract library locked at v4.4.2, while
+ * HoprChannels contract imports OpenZeppelin contract of v4.8.3, it's not possible to import both contracts
+ * the same time without creating conflicts. Therefore, two method identifiers of HoprToken contract are
+ * defined with value instead of `.selector`
+ */
+library HoprCapabilityPermissions {
+    using TargetUtils for Target;
+    using EnumerableTargetSet for TargetSet;
+
+    // HoprChannels method ids (TargetType.CHANNELS)
+    bytes4 internal constant REDEEM_TICKET_SELECTOR = HoprChannels.redeemTicketSafe.selector;
+    bytes4 internal constant CLOSE_INCOMING_CHANNEL_SELECTOR = HoprChannels.closeIncomingChannelSafe.selector;
+    bytes4 internal constant INITIATE_OUTGOING_CHANNEL_CLOSURE_SELECTOR = HoprChannels.initiateOutgoingChannelClosureSafe.selector;
+    bytes4 internal constant FINALIZE_OUTGOING_CHANNEL_CLOSURE_SELECTOR = HoprChannels.finalizeOutgoingChannelClosureSafe.selector;
+    bytes4 internal constant FUND_CHANNEL_SELECTOR = HoprChannels.fundChannelSafe.selector;
+    bytes4 internal constant SET_COMMITMENT_SELECTOR = HoprChannels.setCommitmentSafe.selector;
+    // HoprToken method ids (TargetType.TOKEN). As HoprToken contract is in production, its ABI is static
+    bytes4 internal constant APPROVE_SELECTOR = hex"095ea7b3"; // equivalent to `HoprToken.approve.selector`, for ABI "approve(address,uint256)"
+    bytes4 internal constant SEND_SELECTOR = hex"9bd9bbc6"; // equivalent to `HoprToken.send.selector`, for ABI "send(address,uint256,bytes)"
```

Comment:
why don't use sth like `ERC20.approve.selector`? pretty sure that there is some sample interface / implementation

Reply:
We can. Need to import IERC20 and IERC777